client/extensions/woocommerce/products/products-variation-types-form/index.js

```diff
@@ -0,0 +1,156 @@
+		description: i18n.translate(
+			'A common {{em}}variation type{{/em}} is color. The {{em}}values{{/em}} would be the colors the product is available in.',
+			{ components: { em: <em /> } }
+		),
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			variations: this.props.variations ? this.props.variations : this.getInitialFields(),
+			isVariation: props.product && 'variable' === props.product.type ? true : false,
+		};
+
+		this.addVariation = this.addVariation.bind( this );
+		this.renderInputs = this.renderInputs.bind( this );
+		this.handleToggle = this.handleToggle.bind( this );
+	}
+
+	getInitialFields() {
+		return [ this.getNewFields() ];
+	}
+
+	getNewFields() {
+		return {
+			type: '',
+			values: [],
+		};
+	}
+
+	handleToggle() {
+		this.setState( ( prevState ) => ( {
+			isVariation: ! prevState.isVariation,
+		} ) );
+	}
+
+	updateType( index, event ) {
+		event.preventDefault();
+		const newValue = event.target.value,
+			updatedVariations = clone( this.state.variations );
+		updatedVariations[ index ] = clone( updatedVariations[ index ] );
+		updatedVariations[ index ].type = newValue;
+		this.setState( { variations: updatedVariations } );
+	}
+
+	updateValues( index, value ) {
+		const updatedVariations = clone( this.state.variations );
+		updatedVariations[ index ] = clone( updatedVariations[ index ] );
+		updatedVariations[ index ].values = value;
+		this.setState( { variations: updatedVariations } );
+	}
+
+	renderInputs( variation, index ) {
+		return (
+			<div key={index} className="products-variation-types-form__fieldset">
+				<FormTextInput
+					placeholder={ i18n.translate( 'Color' ) }
+					value={ variation.type }
+					name="type"
+					onChange={ this.updateType.bind( this, index ) }
+					className="products-variation-types-form__field"
+				/>
+				<TokenField
+					placeholder={ i18n.translate( 'Comma separate these' ) }
+					value={ variation.values }
+					name="values"
+					onChange={ this.updateValues.bind( this, index ) }
+				/>
+			</div>
+		);
+	}
+
+	addVariation( event ) {
+		event.preventDefault();
+		const updatedVariations = this.state.variations.concat( [ this.getNewFields() ] );
+		this.setState( { variations: updatedVariations } );
+	}
+
+	render() {
+		const fields = this.state.variations,
+			inputs = fields.map( this.renderInputs ),
+			isNewProduct = this.props.product ? false : true;
+
+		let variationsForm = null;
+		if ( this.state.isVariation ) {
+			variationsForm = (
+				<div className="products-variation-types-form__wrapper">
+					<strong>{ this.props.label }</strong>
+					<p>{ this.props.description }</p>
+
+					<div className="products-variation-types-form__group">
+						<div className="products-variation-types-form__labels">
+							<FormLabel className="products-variation-types-form__label">{ i18n.translate( 'Variation type' ) }</FormLabel>
+							<FormLabel>{ i18n.translate( 'Values' ) }</FormLabel>
```

Comment:
Hi! I've found a possible matching string that has already been translated **90** times:
`translate( 'Value' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).